### PR TITLE
adding first travel page

### PR DESCRIPTION
**Pull request by joeyymoore**
### Description

- Creating a new TravelEntertainmentModel containing the travel and entertainment section fields
- Updating the BenefitsViewModel to take in the new TravelEntertainmentModel
- Creating the new TravelOrEntertainment page
- Linking the new page to the CYA page

Add a link to the relevant story in Jira
https://jira.tools.tax.service.gov.uk/browse/SASS-1376

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of master?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [x]  Have you checked for merge conflicts or any changes in the current master that may affect the current pull request? i.e. does it need another rebase?
- [x]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [x]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of master?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [ ]  Have you checked the PR Builder passes?

### CODE REVIEW

File: app/models/employment/TravelEntertainmentModel.scala [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    travelEntertainmentQuestion match {
+      case Some(true) =>
+        (travelSectionFinished,personalIncidentalSectionFinished,entertainingSectionFinished) match {
+          case (call@Some(_), _, _) => call
+          case (_, call@Some(_), _) => call
+          case (_, _, call@Some(_)) => call
+          case _ => None
+        }
+      case Some(false) => None
+      case None => Some(CheckYourBenefitsController.show(taxYear, employmentId)) //TODO update to my new page
```
Comment (ina-tu) [inline app/models/employment/TravelEntertainmentModel.scala:44]: this TODO can be completed

File: it/controllers/benefits/TravelOrEntertainmentControllerISpec.scala [hunk 2]
```diff
@@ -0,0 +1,492 @@
...
+        "has an SEE_OTHER(303) status" in {
+          result.status shouldBe SEE_OTHER
+          result.header("location") shouldBe Some(s"http://localhost:11111/income-through-software/return/$taxYear/view")
+
+        }
+      }
+
+      "redirect to the check employment benefits page when theres no CYA data" which {
+
+        lazy val form: Map[String, String] = Map(YesNoForm.yesNo -> YesNoForm.yes)
```
Comment (ina-tu) [inline it/controllers/benefits/TravelOrEntertainmentControllerISpec.scala:292]: this form can be removed as it's not used

File: it/controllers/benefits/TravelOrEntertainmentControllerISpec.scala [hunk 3]
```diff
@@ -0,0 +1,492 @@
...
+  }
+
+  val userScenarios: Seq[UserScenario[CommonExpectedResults, SpecificExpectedResults]] = {
+    Seq(UserScenario(isWelsh = false, isAgent = false, CommonExpectedEN, Some(ExpectedIndividualEN)),
+      UserScenario(isWelsh = false, isAgent = true, CommonExpectedEN, Some(ExpectedAgentEN)),
+      UserScenario(isWelsh = true, isAgent = false, CommonExpectedCY, Some(ExpectedIndividualCY)),
+      UserScenario(isWelsh = true, isAgent = true, CommonExpectedCY, Some(ExpectedAgentCY)))
+  }
+
+  object CyaModel {
```
Comment (ina-tu) [inline it/controllers/benefits/TravelOrEntertainmentControllerISpec.scala:126]: this CyaModel can be removed as it's not used

File: it/controllers/benefits/TravelOrEntertainmentControllerISpec.scala [hunk 4]
```diff
@@ -0,0 +1,492 @@
...
+          cyamodel.employment.employmentBenefits.flatMap(_.travelEntertainmentModel.flatMap(_.travelAndSubsistenceQuestion)) shouldBe None
+          cyamodel.employment.employmentBenefits.flatMap(_.travelEntertainmentModel.flatMap(_.travelAndSubsistence)) shouldBe None
+          cyamodel.employment.employmentBenefits.flatMap(_.travelEntertainmentModel.flatMap(_.personalIncidentalExpensesQuestion)) shouldBe None
+          cyamodel.employment.employmentBenefits.flatMap(_.travelEntertainmentModel.flatMap(_.personalIncidentalExpenses)) shouldBe None
+          cyamodel.employment.employmentBenefits.flatMap(_.travelEntertainmentModel.flatMap(_.entertainingQuestion)) shouldBe None
+          cyamodel.employment.employmentBenefits.flatMap(_.travelEntertainmentModel.flatMap(_.entertaining)) shouldBe None
+        }
+      }
+
+      "redirect the user to the overview page when it is in year" which {
```
Comment (ina-tu) [inline it/controllers/benefits/TravelOrEntertainmentControllerISpec.scala:454]: should this test and the one below it be outside of the "redirect to another page when a valid request is made and then" tests?

Comment (joeyymoore) [reply it/controllers/benefits/TravelOrEntertainmentControllerISpec.scala:454]: So the "redirect to another page when a valid request is made and then" might be misleading. This just kinda means any tests where we redirect away from the current page. I wasn't sure how to word it.

File: it/controllers/benefits/TravelOrEntertainmentControllerISpec.scala (hunk 3 above)
Comment (joeyymoore) [reply it/controllers/benefits/TravelOrEntertainmentControllerISpec.scala:126]: removed

**Pull request merged**